### DISCUSSION TITLE
Intercom, channels and communication means

### DISCUSSION THREAD

**Discussion by ViktorShield** [FakeFishGames/Barotrauma#9444] [Ideas and suggestions]
There's a command staff, medical personnel, repairs department and a security team. If all of them will use the same channel to communicate, it will become hard to. But when using several channels there's another issue - you cannot give/receive orders from those on other channels without switching. So, if I am a captain, I have to switch channel every time I want to contact medics, for example. Even more, I have to say the same thing over and over again in different channels to make a global message. It's not very convenient, I must say.

Solution?
SS13-like radio system. The problems mentioned above are somewhat already solved in it. But all of the communication there is via text messages, so, we need something else

There'll be intercoms. Anyone(or with a specific ID card) can click it and broadcast a message sub-wide to a specific department or anyone regardless of a channel.

Pros:
- No more chaos in channel 0
- A department will find it easier to communicate within this department since they're not interrupting anyone and can focus on their duty. For example, a mechanic won't hear 2 medics desperately trying to perform a surgery. And medics are free to discuss whatever medics can discuss, and no one will ask them not to
- Captain can give loud sub-wide orders without being interrupted in any way.
- Captain can give department-speficic orders without being interrupted and with no need to swap channels
- More intuitive(just click the channels you want to hear from and then the channels you want to broadcast in)
- More roleplay 

_[English is not my native lang, I tried my best to express what I wanted to]_

**Comment by Jinan-Dangor** [comment]
I second this. 70% of stuff you need to do with comms can be done through local and global, but radio channels fix just as many problems as they solve unless you have an intercom that can broadcast onto a spectrum (because now the medics talking on the medic channel can't hear the captain screaming for them to save someone). Even just having a 'broadcast headset' where you set a min and max channel and broadcast your voice onto all the channels in that range would do wonders (the captain could start with this headset while everyone else starts with basic ones).

**Comment by grufkork** [comment]
Been thinking about this and looked at ways to implement it. To make it fairly realistic and without breaking the current rules of radio communication in game, being able to scan a set of frequencies could work well. To be able to communicate in the manner described above, you either need single-transmit/multi-listen OR multi-transmit/single-listen. Multi-listen generally works better as you choose what channels to subscribe to, rather than the sender having to choose whom/what to transmit to. 

A nice, hopefully not-too-breaking way to do this would be channel scanning. A certain number of times per second, the player’s radio jumps between a preset list of channels. When a transmission on a channel is detected, it stops on that channel until the transmission ends, then returns to scanning. This of course means you might miss simultaneous transmissions on different channels, but the upside is that Wifi components don’t suddenly need to be able to listen to multiple channels. It also causes a bit of a nerf and makes multichannel comms more of a puzzle to manage, and less of a straight-out improvement. (It is also more realistic, but *that* isn’t necessarily better…) To send, you could map keys to different channels, or maybe just have one key to temporarily switch to the chosen broadcast channel (depending on how scaleable it should be). 

A problem could be for channels with lots of chatter blocking people from receiving on other channels. Requiring radio discipline could be a bit too hardcore/unfun in some groups. Then again, it wouldn’t impact current usage of radios but only extend the possibilities. And while Barotrauma isn’t ultra-hardcore, it has a lot of depth and detail to explore that isn’t immediately graspable. 

One implementation question is how many channels should be available to switch between. Just having two opens up a lot of possibilities, and if there’s *LOTS* of people, you could have a group leaders acting relays/switchboards to further structure communications, which you’ll probably need anyways if you are that many. But there’s also the question whether this goes against Barotrauma’s design goals, particularly the social part. Splitting people into groups means less interactions and part of the communication/coordination puzzle being solved automatically. Then again, the short while my crew tried using separate channels mutual trust between departments quickly collapsed and spying ensued.
